### PR TITLE
Add some actions and conditions to bots

### PR DESCRIPTION
**Pull request by Gireen**

### CODE REVIEW

File: src/sgame/sg_bot_ai.cpp [hunk 1]
```diff
@@ -621,10 +621,25 @@ AINodeStatus_t BotActionClassDodge( gentity_t *self, AIGenericNode_t* )
...
+		botEntityAndDistance_t e = AIEntityToGentity( self, et );
+		if ( !BotChangeGoalEntity( self, e.ent ) )
+		{
+			return STATUS_FAILURE;
+		}
+	}
+	else if( a->nparams == 3 )
+	{
+		vec3_t pos = { AIUnBoxFloat(a->params[0]), AIUnBoxFloat(a->params[1]), AIUnBoxFloat(a->params[2]) };
+		if ( !BotChangeGoalPos( self, pos ) )
```
Comment (slipher) [inline src/sgame/sg_bot_ai.cpp:645]: What is the use case of this?

File: src/sgame/sg_bot_ai.cpp [hunk 2]
```diff
@@ -540,6 +540,14 @@ AINodeStatus_t BotActionFireWeapon( gentity_t *self, AIGenericNode_t* )
 	return STATUS_SUCCESS;
 }
 
+AINodeStatus_t BotActionTeleport( gentity_t *self, AIGenericNode_t *node )
```
Comment (slipher) [inline src/sgame/sg_bot_ai.cpp:543]: Is this for debugging or what?

File: src/sgame/sg_bot.h [hunk 3]
```diff
@@ -92,6 +92,8 @@ struct botMemory_t
 	int lastThink;
 	int stuckTime;
 	vec3_t stuckPosition;
+
+	int aliveTime;
```
Comment (slipher) [inline src/sgame/sg_bot.h:96]: `spawnTime` would be more descriptive

File: src/sgame/sg_admin.cpp [hunk 4]
```diff
@@ -5733,6 +5733,22 @@ bool G_admin_bot( gentity_t *ent )
 			G_BotDel( clientNum ); //delete the bot
 		}
 	}
+	else if ( !Q_stricmp( arg1, "behavior" ) && args.Argc() == 4 )
```
Comment (slipher) [inline src/sgame/sg_admin.cpp:5736]: Add the new subcommand to `BotUsage`

File: src/sgame/sg_admin.cpp [hunk 5]
```diff
@@ -5733,6 +5733,22 @@ bool G_admin_bot( gentity_t *ent )
...
+
+		char err[MAX_STRING_CHARS];
+		const char *name = args[2].data();
+		int clientNum = G_ClientNumberFromString( name, err, sizeof( err ) );
+		if ( clientNum == -1 ) //something went wrong when finding the client Number
+		{
+			ADMP( va( "%s %s %s", QQ( "^3$1$:^* $2t$" ), "bot", Quote( err ) ) );
+			return false;
+		}
+		const char *bahavior = args[3].data();
```
Comment (slipher) [inline src/sgame/sg_admin.cpp:5748]: spelling

File: src/sgame/sg_bot.cpp [hunk 6]
```diff
@@ -205,7 +205,28 @@ bool G_BotSetDefaults( int clientNum, team_t team, int skill, const char* behavi
 			return false;
 		}
 	}
+	return true;
```
Comment (slipher) [inline src/sgame/sg_bot.cpp:208]: If you are changing the behavior of an existing bot, don't you need to do the stuff with clearing the current/running nodes?

**Comment by Gireen** [comment]
Teleporting or vectors as goals can be used for testing but also any other "scripted" actions like on mission maps or in a tutorial.

File: src/sgame/sg_bot.cpp (hunk 6 above)
Comment (Gireen) [reply src/sgame/sg_bot.cpp:208]: I don't know. It seems to work without, but you are probably right. 😄

**Comment by illwieckz** [comment]
> Teleporting or vectors as goals can be used for […] tutorial.

That's good!

**Comment by illwieckz** [comment]
Oh, I forgot about that PR, what's missing for merging?

File: src/sgame/sg_bot.h (hunk 3 above)
Comment (Gireen) [reply src/sgame/sg_bot.h:96]: im not sure but i think alivetime matches better.
the time a bot is alive. like stucktime is the time that a bot is well stuck.
spawntime could maybe also be misunderstood as time it takes to spawn or similar.

Comment (illwieckz) [reply src/sgame/sg_bot.h:96]: I agree with `aliveTime` being more explicit.

Comment (slipher) [reply src/sgame/sg_bot.h:96]: aliveTime is good for the function which can be used in the behavior tree - amount of time the bot has been alive. I think spawnTime is better just for this variable, which contains the time the bot spawned.

Comment (Gireen) [reply src/sgame/sg_bot.h:96]: oh yes. that's absolute correct. 😲

**Comment by slipher** [comment]
Everything looks good!

**Comment by illwieckz** [comment]
@Gireen, I believe the behavior tree does not make use of the feature to delay bot evolve yet?

**Pull request merged**